api/src/main/java/org/openmrs/DrugOrder.java

```diff
@@ -547,6 +549,13 @@ public String getDrugNonCoded() {
 	 */
 	public void setDrugNonCoded(String drugNonCoded) {
 		this.drugNonCoded = StringUtils.isNotBlank(drugNonCoded) ? drugNonCoded.trim() : drugNonCoded;
+		if (StringUtils.isNotBlank(drugNonCoded) && getConcept() == null) {
+			String conceptUuid = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GP_DRUG_ORDER_DRUG_OTHER);
+			if (StringUtils.isNotBlank(conceptUuid)) {
```

Comment:
I am not a fun of having Context.getService methods in our domain objects. Why not just set this at the service layer only?